File: 1512-number-of-good-pairs/1512-number-of-good-pairs.java
class Solution {
    public int numIdenticalPairs(int[] nums) {
        int identicalPairs = 0;
        int i = 0, j = 1;
        while (i < nums.length - 1) {
            if (nums[i] == nums[j]) {
                identicalPairs++;
                j++;
            } else {
                j++;
            }
            if (j == nums.length) {
                i++;
                j = i + 1;
            }
        }
        return identicalPairs;
    }
}

File: 1512-number-of-good-pairs/1512-number-of-good-pairs.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var numIdenticalPairs = function(nums) {
  let pairCount = 0;
  for (let i = 0; i <= nums.length - 1; i++) {
    let j = i + 1;
    let currentNum = nums[i];
    let potentialMatchingNum = nums[j];
    while (j <= nums.length - 1) {
      potentialMatchingNum = nums[j];
      if (currentNum === potentialMatchingNum) {
        pairCount++;
      }
      j++;
    } 
  }
  return pairCount;
}